surefire-providers/surefire-junit47/src/main/java/org/apache/maven/surefire/junitcore/ConcurrentRunListener.java

```diff
@@ -61,10 +60,10 @@
         this.reportImmediately = reportImmediately;
         this.classMethodCounts = classMethodCounts;
         this.consoleStream = consoleStream;
-        reporterManagerThreadLocal = new ThreadLocal<RunListener>()
+        reporterManagerThreadLocal = new ThreadLocal<TestRunListener>()
         {
             @Override
-            protected RunListener initialValue()
+            protected TestRunListener initialValue()
             {
                 return reporterFactory.createReporter();
             }
```

Comment:
now can be like:
```
 reporterManagerThreadLocal = ThreadLocal.withInitial( reporterFactory::createReporter );

```

TestSetFailedException is not thrown from this constructor - line 58

Reply:
done